playbooks/nfd_operator_presubmit.yml

```diff
@@ -0,0 +1,8 @@
+---
```

Comment:
This is not ok, I am working on rewritting the repo to follow Ansible best practices, as it is today, the repo simply is not a prod lvl Ansible repo, and that is leading to duplicated code and the repo is becoming more and more complex to maintain , which is the oposite from Ansible philosophy

Reply:
the only thing I suggest is to duplicate the *playbook*, nothing else here

> the repo simply is not a prod lvl Ansible repo

what part are you thinking about?

I'm looking forward for suggestions and PRs to remove duplicated code, I've been struggle it with it as well: how to keep functionalities grouped by purpose, with clearly defined parameters (`defaults/` directory) vs avoiding code duplication (eg, the code required to clone&build a git repo and push it to an image stream) ...

Reply:
I'm thinking what Eduardo means is that having a single role per playbook means we have zero reusability. I believe he's aiming for us to have tiny roles to be used and combined in playbooks like Legos, rather than having playbooks be an empty wrapper around giant roles.

Reply:
yes, that could make sense, but I'm curious to see what it could give, I don't find roles modular enough to encourage reuse. 

I think there's a concept of "function", eg a task file that can be called on demand, but I don't remember the exact ansible name

there's also the concept of TAGS (https://github.com/openshift-psap/ci-artifacts/issues/167) that seemed useful when I followed an ansible course, but I didn't dare trying to use it yet

> I would recommend keeping one role per playbook
with modularity in mind, then the advise would be: keep one "high level functionality" per playbook, hopefully composed of multiple "lower-level roles"